pkg/idms/idms.go

```diff
@@ -0,0 +1,263 @@
+package idms
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	configv1 "github.com/openshift/api/config/v1"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtimeClient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// IDMSBuilder provides struct for configmap object containing connection to the cluster and the configmap definitions.
+type IDMSBuilder struct {
+	// ConfigMap definition. Used to create configmap object.
+	Definition *configv1.ImageDigestMirrorSet
+	// Created configmap object.
+	Object *configv1.ImageDigestMirrorSet
+	// Used in functions that defines or mutates configmap definition. errorMsg is processed before the configmap
+	// object is created.
+	errorMsg  string
+	apiClient runtimeClient.Client
+}
+
+// NewIDMSBuilder creates a new instance of IDMSBuilder.
+func NewIDMSBuilder(apiClient *clients.Settings, name string, mirror configv1.ImageDigestMirrors) *IDMSBuilder {
+	glog.V(100).Infof(
+		"Initializing new imagedigestmirrorset structure with the following params: "+
+			"name: %s, mirror: %v", name, mirror)
+
+	if apiClient == nil {
+		glog.V(100).Infof("The apiClient cannot be nil")
+
+		return nil
+	}
+
```

Comment:
It won't work at the moment because there are other resources using the same scheme which is being added here https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/clients/clients.go#L288